src/Raven.Client/Documents/LinqExtensions.cs

```diff
@@ -829,27 +830,39 @@ public static Task<TSource> SingleOrDefaultAsync<TSource>(this IQueryable<TSourc
             return query.SingleOrDefaultAsync(token);
         }
 
-        /// <summary>
+        /// <summary>        
         /// Perform a search for documents which fields that match the searchTerms.
-        /// If there is more than a single term, each of them will be checked independently.
+        /// If there is more than a single term, each of them will be checked independently.        
         /// </summary>
+        /// <typeparam name="T">The type of element of self</typeparam>
+        /// <param name="self">The <see cref="IQueryable{T}"/> to search on</param>
+        /// <param name="fieldSelector">Function returning the field to search on</param>
+        /// <param name="searchTerms">Field terms to search for, seperated with whitespaces</param>
+        /// <param name="boost">Boost factor for sorting purposes</param>
+        /// <param name="options">Logical operator to use in relation to the previous filtering statement.</param>
+        /// <param name="termsSearchOperator">Determines the logical operator between all of the terms received in searchTemrs parameter</param>
+        /// <returns></returns>
         public static IRavenQueryable<T> Search<T>(this IQueryable<T> self, Expression<Func<T, object>> fieldSelector, string searchTerms,
```

Comment:
You need to create an overload here, not modify the method.
This is a backward compatability concern